### PR TITLE
add modifiers to render tree

### PR DESCRIPTION
**Pull request by patricklx**
## Description
this adds modifiers and their html element to the render tree.

this also improves the UX for positional args, especially functions

this builds on #2549

### CODE REVIEW

**Comment by patricklx** [comment]
image

**Comment by patricklx** [comment]
this is ready

**Review by RobbieTheWagner — DISMISSED** [review]
LGTM!

**Comment by RobbieTheWagner** [comment]
@patricklx the code and tests look great here, thank you for taking this on! I was looking at the screenshot though, and shouldn't the modifier be inside the element? They seem to be outside of the closing `>`.

**Comment by patricklx** [comment]
It's by design, i didn't try to put it inside. Maybe it can be done,  i can have a look tomorrow

**Comment by patricklx** [comment]
I think it makes more sense to have it under the element for ux. Otherwise it can be too much on one line.

**Comment by patricklx** [comment]
@RobbieTheWagner i tried a few things, but was not able to do it in a good way:
maybe add some indicator that its inside an element? 

Screenshot 2024-02-09 at 14 22 33

**Comment by patricklx** [comment]
Wrapping is not good in this case, since we are using vertical-collection which needs predictable row heights. 
So everything would need to be in one line...


Also
i often prefer to have it like this:
```
<li
    class="foo"
   {{did-insert this.myCoolFunction}}
>
```
:)

**Comment by patricklx** [comment]
how about this:
Screenshot 2024-02-09 at 17 19 23

**Comment by RobbieTheWagner** [comment]
> how about this: Screenshot 2024-02-09 at 17 19 23

This looks great! 👍

**Comment by patricklx** [comment]
pushed the changes

**Comment by patricklx** [comment]
@RobbieTheWagner what about this?

File: skeletons/web-extension/content-script.js [hunk 1]
```diff
@@ -24,6 +24,7 @@
    *     https://developer.mozilla.org/en-US/docs/Web/API/Channel_Messaging_API
    */
   window.addEventListener('message', function(event) {
+    console.log('message', event);
```
Comment (RobbieTheWagner) [inline skeletons/web-extension/content-script.js:27]: Do we want to keep all of these console logs or were they for testing?

Comment (patricklx) [reply skeletons/web-extension/content-script.js]: ready

**Comment by patricklx** [comment]
I found that this might need more work. The modifiers need to be removed from debug render tree when they get destroyed

**Comment by patricklx** [comment]
@RobbieTheWagner ready

**Comment by RobbieTheWagner** [comment]
@patricklx looks like there are some conflicts. Would you mind resolving those please?

**Comment by patricklx** [comment]
@RobbieTheWagner rebased

**Review by RobbieTheWagner — APPROVED** [review]

**Pull request merged**